pkg/image/registry/imagestreammapping/rest.go

```diff
@@ -125,3 +125,41 @@ func (s *REST) findStreamForMapping(ctx kapi.Context, mapping *api.ImageStreamMa
 	}
 	return nil, errors.NewNotFound("ImageStream", "")
 }
+
+// Delete deletes the given image and the image stream tag associated with it
+func (s *REST) Delete(ctx kapi.Context, id string) (runtime.Object, error) {
+	streamName, tag, _ := api.SplitImageStreamTag(id)
+
+	// Get image stream
+	stream, err := s.imageStreamRegistry.GetImageStream(ctx, streamName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Find associated image name via image stream tags
+	var image string
+	if stream.Status.Tags != nil {
+		eventList, ok := stream.Status.Tags[tag]
+		if !ok {
+			return &kapi.Status{Status: kapi.StatusSuccess}, nil
+		}
+		// Check if list is empty? Sort based on timestamp?
+		image = eventList.Items[0].Image
+	}
+	if len(image) == 0 {
+		return &kapi.Status{Status: kapi.StatusSuccess}, nil
+	}
+
+	// Delete image
+	if err := s.imageRegistry.DeleteImage(ctx, image); err != nil {
```

Comment:
Do not delete the image. Only delete the tag. `oadm prune images` will delete the image, if appropriate.